added first draft SingleValueEncodingContainer doc (#3033)
Codecademy/docs: content/swift/concepts/protocols/SingleValueEncodingContainer.md

---

> Title: 'SingleValueEncodingContainer Protocols'

Reviewer: ```suggestion
Title: 'SingleValueEncodingContainer'
```

Description: 'A container that can support the storage and direct encoding of a single non-keyed value.'
Subjects:
  - 'iOS'
  - 'Mobile Development'
Tags:
  - ['Protocols']()
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

A **SingleValueEncodingContainer** protocol is a container that can support the storage and direct encoding of a single non-keyed value. It is part of Swift's `Encoder` framework.

> The values you can encode will depend on the chosen encoding format (JSON, plist, etc.).

Reviewer: ```suggestion
The values that can be encoded will depend on the chosen encoding format (JSON, plist, etc.).
```

## Syntax

```psuedo
protocol SingleValueEncodingContainer {
  // Protocol body
}

> ```

Reviewer: We typically have a sentence explaining the syntax.

## Example

> 

Reviewer: You need a introductory sentence summarizing the example.

```swift
import Foundation

struct Person: Encodable {
    let name: String
    let age: Int
    
    func encode(to encoder: Encoder) throws {
        var container = encoder.singleValueContainer()
        
        try container.encode(name) // Encoding the name
        // You can also encode the age or any other single value property here
    }
}

let person = Person(name: "Alexandria", age: 30)

let encoder = JSONEncoder()
do {
    let jsonData = try encoder.encode(person)
    if let jsonString = String(data: jsonData, encoding: .utf8) {
        print(jsonString)
    }
} catch {
    print(error)
}

> ```

Reviewer: You need a ` ```shell` block following the example showing the output, like this:

````
This example results in the following output:

```shell
The example's output
```
````

> 
> Official Swift Documentation: https://developer.apple.com/documentation/swift/singlevalueencodingcontainer

Reviewer: ```suggestion
```

We don't like to link to off-site documentation in Docs.
